### PR TITLE
Implement watchonly for fundrawtransaction

### PR DESCRIPTION
**Pull request by kanzure**
Support watchonly in fundrawtransaction, CreateTransaction, SelectCoins and coin selection in general.

Coin selection is exposed over RPC through fundrawtransaction, and watchonly can be enabled by passing includeWatching true (defaults to false).

fundrawtransaction will first attempt to fund a transaction without using watchonly, even when includeWatching is enabled. When this fails, an includeWatching attempt is made.

When an includeWatching attempt at coin selection is performed (in CreateTransaction), and all watchonlys are consumed leaving no funds available for a fee, then CreateTransaction will not include a fee in the transaction.

Also, includeWatching tests (for fundrawtransaction) added here are passing.

Reviewers! Here are some things I would especially appreciate eyelooking for: suggestions for more edge cases for the tests, other RPC commands that should be tested, other tests for other features that I should double check for whether or not this makes any impacts, locations or existence of any sort of documentation I should be updating, unintended side effects introduced in CreateTransaction like by multiple parameters that may eventually be used simultaneously in ways that I am failing to anticipate....

I received a few suggestions from others that SignSignature should be replaced with an estimator in CreateTransaction. However, this estimator apparently does not seem to be necessary for implementing watchonly-support here, so this suggestion may go into another ticket instead.

This requires #5503 to be merged (I wouldn't mind this getting merged into #5503).

### CODE REVIEW

**Comment by kanzure** [comment]
> Breaks travis

Tests are passing now.

**Comment by jonasschnelli** [comment]
@kanzure could you rebase on top of #5503?

Conceptual ACK.
Could also go into 0.11 together with #5503 IMO.

**Comment by kanzure** [comment]
Rebased on to the fundrawtransaction branch. Travis is reporting an error:

```
Tests successful
Running testscript rawtransactions.py...
Initializing test directory /tmp/testZdbica
JSONRPC error: Expected type int, got str
  File "/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/test_framework.py", line 117, in main
    self.run_test()
  File "/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/rawtransactions.py", line 68, in run_test
    rawtxfund = self.nodes[2].fundrawtransaction(rawtx)
  File "/home/travis/build/bitcoin/bitcoin/bitcoin-x86_64-unknown-linux-gnu/qa/rpc-tests/python-bitcoinrpc/bitcoinrpc/authproxy.py", line 126, in __call__
    raise JSONRPCException(response['error'])
Cleaning up
```

File: src/rpcrawtransaction.cpp [hunk 1]
```diff
@@ -766,3 +766,59 @@ Value sendrawtransaction(const Array& params, bool fHelp)
...
+                            + HelpExampleCli("fundrawtransaction", "\"myhex\"") +
+                            "\nSign the transaction, and get back the hex\n"
+                            + HelpExampleCli("signrawtransaction", "\"myhex\"") +
+                            "\nSend the transaction (signed hex)\n"
+                            + HelpExampleCli("sendrawtransaction", "\"signedhex\"") +
+                            "\nAs a json rpc call\n"
+                            + HelpExampleRpc("sendrawtransaction", "\"signedhex\"")
+                            );
+    
+    RPCTypeCheck(params, boost::assign::list_of(int_type)(int_type)(array_type));
```
Comment (dexX7) [inline src/rpcrawtransaction.cpp]: Should probably be `RPCTypeCheck(params, boost::assign::list_of(str_type)(bool_type));`.

**Comment by TheBlueMatt** [comment]
Why try to fund without watchonly first? If I call the RPC and say I want to include watchonly, it means I want to include watchonly (and get the best fee from the inputs I opted to select).

**Comment by jonasschnelli** [comment]
I'd like to add this to #5503 but the nFeeRet=0 as well as the issue mentioned by @TheBlueMatt (calling `CreateTransaction()` two times) are unclear and needs fixing. RPC tests should also send a funded raw transaction and check the response to make sure it passes the mempool checks (i have a feeling that it won't because of the generated fee)

**Comment by kanzure** [comment]
Closed in favor of #6088.

**Pull request closed without merging**